DOCSP-14352: fix multikey index description & example (#542)
mongodb/docs-node: source/fundamentals/indexes.txt

Multikey Indexes (Indexes on Array Fields)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> **Multikey indexes** are indexes that improve performance for queries on

Reviewer: ```suggestion
**Multikey indexes** are indexes that improve the performance of queries on
```

fields that contain array values. You can define a multikey index using

> the same syntax as when you create a single field or compound index.

Reviewer: Suggestion:
I wonder if the second sentence here is necessary since you show the syntax. The only value I can see is if a reader had just read the prior two sections, but then might have to refer back to them to understand what that syntax was. I would probably omit that sentence.

> The following example uses the ``createIndex()`` method to create an ascending

Reviewer: Suggestion: 
I think "calls" could be more precise in describing what the example does with the createIndex() method.

```suggestion
The following example calls the ``createIndex()`` method to create an ascending
```

index on the ``cast`` field, which contains an array of names, in the

> ``movies`` collection of the ``sample_mflix`` database.

Reviewer: Suggestion:

From reading the current content, I think introducing the method could be separated out into its own sentence rather than combining it with a description of the example. Seems like this uses the same pattern as the Single Field Index section, but maybe that should be separated as well.

Personal preference, but I try to avoid separating out non-restrictive phrases with commas or parentheses in docs because it slows down my own reading speed. I wasn't able to think of a good way to phrase this, so I left it to the reader to infer the cast field is an array of values in the following suggested edit:

"You can create a multikey index on a field that contains an array of values by calling the createIndex() method. The following example shows how you can create an ascending index on the cast field in the movies collection of the sample_mflix database".

.. literalinclude:: /code-snippets/indexes/multikey.js
   :language: js
   :start-after: begin-idx
   :end-before: end-idx
   :dedent:

> The following code performs a query that uses the index on
> the ``cast`` field:

Reviewer: Suggestion:
I think the previous wording is more precise than "uses", but here's an alternative that is less wordy that could work:

```suggestion
The following code queries the multikey index on the ``cast`` field
```
